### DISCUSSION TITLE
Using docstring as field description in json schema

### DISCUSSION THREAD

**Discussion by guillaume-alvarez** [pydantic/pydantic#10106] [Question] [answered]
Hi!

I noticed that while pydantic uses the class docstring as description when generating a json schema, it does not uses the field docstrings when generating their schema.

I can set the description in a `Field` object: `Field(..., description="my description")` but the IDE support is not the same as with a docstring.

Is it a known limitation? An error in the way I describe fields?

**Comment by Viicos** [comment, marked as answer]
This was implemented in Pydantic 2.7 with the [`use_attribute_docstrings`](https://docs.pydantic.dev/latest/api/config/#pydantic.config.ConfigDict.use_attribute_docstrings) configuration value.

This wasn't made opt-in for various reasons:
- Backwards compatibility
- Potential bad performance (we inspect the AST to extract those docstrings)
- Potential limitations in very specific scenarios.

**Reply by guillaume-alvarez** [reply]
Thanks a lot!
I guess the performance impact is limited to the init of the process when the module in which the model is defined is loaded?

**Reply by Viicos** [reply]
Exactly, it is only at class definition. So this hopefully won't be an issue unless you're having hundreds of models defined.
